### PR TITLE
feat(websocket): MED-173 replace websockets with api call in favorite page

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- replace websocket init and update calls
- add document_types in signet model to enable filter

new config:
```
"mediacentreUpdateFrequency" : ${mediacentreUpdateFrequency}
```

new dependencies:
```
"@types/babel__traverse": "7.14.2",
"axios-mock-adapter": "^1.19.0"
```

## Checklist tests
use "favorites" page as usual

## Issue ticket number and link
[MED-173](https://jira.support-ent.fr/browse/MED-173)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "@types/jquery": "^2.0.34",
     "@types/angular": "^1.6.54",
     "@types/core-js": "0.9.42",
-    "typescript": "^3.5.2",
+    "@types/babel__traverse": "7.14.2",
```
Comment (alicedraillard) [inline package.json:35]: used to enable node compilation

File: src/main/resources/i18n/en.json [hunk 2]
```diff
@@ -1,3 +1,142 @@
 {
-  "mediacentre.title": "Mediacentre"
+  "mediacentre.title": "Mediacentre",
```
Comment (alicedraillard) [inline src/main/resources/i18n/en.json:2]: Add english i18n

File: src/main/resources/public/ts/controllers/favorite.ts [hunk 3]
```diff
@@ -1,105 +1,115 @@
...
+        this.updateFrequency = mediacentreUpdateFrequency;
+
+        await this.fetchFavorites();
+
+        let viewModel: IViewModel = this;
+        this.$scope.$on('deleteFavorite', function (event, id) {
+            viewModel.favorites = viewModel.favorites.filter(el => el.id !== id);
+        });
+
+        this.$interval(async (): Promise<void> => {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/favorite.ts:64]: replace websocket with api call in interval

File: src/main/resources/public/ts/controllers/favorite.ts [hunk 4]
```diff
@@ -1,105 +1,115 @@
...
+            await this.fetchFavorites();
+        }, this.updateFrequency, 0, false);
+    }
+
+    showFilter(): void {
+        this.displayFilter = !this.displayFilter;
+    }
 
-    initFavorite();
+    async fetchFavorites(filteredResources?: Resource[]) {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/favorite.ts:73]: optional parameters used to keep instance when using directive multi-combo (cf [this example](https://github.com/OPEN-ENT-NG/vie-scolaire/blob/dev/src/main/resources/public/modules/viescolaire/directives/timeslotAudience/timeslot-audience.ts#L101))

File: src/main/java/fr/openent/mediacentre/controller/MediacentreController.java [hunk 5]
```diff
@@ -56,7 +56,8 @@ public void view(HttpServerRequest request) {
         JsonObject params = new JsonObject()
                 .put("wsPort", "dev".equals(config.getString("mode")) ? Mediacentre.wsPort : "")
                 .put("mode", config.getString("mode"))
-                .put("sources", sourceList);
+                .put("sources", sourceList)
+                .put(Field.MEDIACENTREUPDATEFREQUENCY, config.getInteger(Field.MEDIACENTREUPDATEFREQUENCY, 60000));
```
Comment (alicedraillard) [inline src/main/java/fr/openent/mediacentre/controller/MediacentreController.java:60]: add config to determine update frequency

**Review by flmartineau — APPROVED** [review]

**Review by c4role — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**